Framework/PythonInterface/plugins/algorithms/LoadEXED.py

```diff
@@ -0,0 +1,189 @@
+            if line.find("Comment")>-1:
+                parms_dict['Comment']=line.strip('Comment =')
+        if line.find("Following are binary data")>0:
+            header=False
+            while line.find("DATA=")<0:
+                b_line = fin.readline()
+                line=b_line.decode('ascii')
+                #print line
+    nrows=int(parms_dict['NDET'])
+    nbins=int(parms_dict['NTC'])
+    print ("read UDET")
+    det_udet = struct_data_read(fin,nrows)
+
+    print ("read Counter")
+    det_count = struct_data_read(fin,nrows)
+
+    print ("read TimeBinBoundaries")
+    det_tbc = struct_data_read(fin,nbins+1,'f')
+
+    print ("read Data")
+    data = np.fromfile(fin, np.uint32, nrows*nbins, '')
+
+    fin.close()
+    parms_dict['phi']=copy.deepcopy(parms_dict['CAR_OMEGA_MAG'])
+    return parms_dict, det_udet, det_count, det_tbc, data
+
+
+def struct_data_read(fin,nrows,data_type='i',byte_size=4):
+    """
+    helper function to read binary data_type
+    requires the file handle, number of rows and data_type
+    """
+    print(nrows)
+    tmp_lst=[struct.unpack(data_type,fin.read(byte_size))[0] for i in range(nrows)]
+    #for i in range(nrows):
+    #    data = struct.unpack(data_type,fin.read(byte_size))[0]
+    #    tmp_lst.append(data)
+    #print(tmp_lst)
+    return tmp_lst
+
+
+class LoadEXED(PythonAlgorithm):
+    __doc__ = """This is the EXED data loader written in Python.
+    Based off code was written by Wolf-Dieter Stein and  Maciej Bartkowiak.
+    Garrett Granroth (ORNL)
+    """
+
+    def category(self):
+        return "Inelastic;Diffraction;PythonAlgorithms"
+
+    def name(self):
+        return "LoadEXED"
+
+    def version(self):
+        return 1
+
+    def summary(self):
+        return """Modifications have been made by Maciej Bartkowiak and Garrett Granroth (ORNL) in 2017,
+        in order to fully replace the C++ version of LoadEXED,
+        improve the treatment of the monitor spectra and Bring Exed into the Mantid Distribution."""
+
+    def PyInit(self):
+        self.declareProperty(FileProperty(name="Filename",defaultValue = "",
+                                          action=FileAction.Load,
+                                          extensions = ["raw"]),
+                             doc="Data file produced by egraph.")
+
+        self.declareProperty(FileProperty(name="InstrumentXML",defaultValue = "",
+                                          action=FileAction.OptionalLoad,
+                                          extensions = ["xml"]),
+                             doc="Instrument definition file. If no file is specified, the default idf is used.")
+
+        self.declareProperty(MatrixWorkspaceProperty(name="OutputWorkspace",
+                                                     defaultValue = "",
+                                                     direction=Direction.Output),
+                             doc="Mantid workspace containing the measured data.")
+
```

Comment:
I don't want to break user scripts at this time.  I will make this an issue for the next release.